orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt

```diff
@@ -60,33 +60,16 @@ interface AuthenticationAware {
   fun retrieveAuthenticatedUser(stage: StageExecution) : PipelineExecution.AuthenticationDetails? {
     return stageNavigator
       .ancestors(stage)
-      .firstOrNull { it.stageBuilder is AuthenticatedStage } ?.let{
-        (it.stageBuilder as AuthenticatedStage).authenticatedUser(solveAuthStages(it.stage)).orElse(null)
+      .firstOrNull {
+        it.stageBuilder is AuthenticatedStage &&
+        it.stage.isManualJudgmentType &&
+        !it.stage.status.isSkipped &&
+        it.stage.withPropagateAuthentication()
+      }?.let{
+        val authStage = it.stage
+        return if(authStage != null) {
+          (it.stageBuilder as AuthenticatedStage).authenticatedUser(authStage).orElse(null)
+        } else null
       }
   }
-
-
-  // When a first valid candidate is found in the ancestors chain is returned
-  // until the ancestor chain was iterated completely at the pipeline beginning
-  fun backtrackStages(stage: StageExecution): StageExecution {
```

Comment:
None of this recursion should be needed - `stageNavigator.ancestors()` provides an ordered list of ancestor stages back to the root, and iterating over that according to the above criteria will find the proper stage to import authentication context from.